src/main/scala/temple/builder/project/ProjectBuilder.scala

```diff
@@ -114,9 +115,18 @@ object ProjectBuilder {
       case (name, service, port) =>
         val serviceRoot =
           ServerBuilder.buildServiceRoot(name, service, port.service, endpoints(service), detail, usesAuth)
-        service.lookupMetadata[ServiceLanguage].getOrElse(ProjectConfig.defaultLanguage) match {
+        val serverFiles = service.lookupMetadata[ServiceLanguage].getOrElse(ProjectConfig.defaultLanguage) match {
           case ServiceLanguage.Go => GoServiceGenerator.generate(serviceRoot)
         }
+
+        val serviceComms = serviceRoot.comms.map { service =>
+          val (_, _, ports) = templefile.providedServicesWithPorts.find { _._1 == service }.get
```

Comment:
I see, maybe worth extracting this to a function in `Templefile`? But it’s only unsafe if the service isn’t really in the templefile